Fix missing dependencies issue during multi-process building

renchenglei opened the pull request:
anv_extensions.h is generated during building. When we build vilkan
with multi-process, libmesa_anv_gen* module may be built befroe the
libmesa_vulkan_common. In this case, we may encounter anv_extensions.h
missing issue.

Let's reorg the generation order, and create anv_extensions.h in
libmesa_anv_entrypoints module. libmesa_anv_gen* includes library
libmesa_anv_entrypoints, which can help avoid this missing
dependencies issue.

Change-Id: I80926d49c0fc2fb857a0ef2da87b674057994d4b
Tests: vulkan build successfully with multi-process
Tracked-On: https://jira.devtools.intel.com/browse/OAM-79750
Signed-off-by: Chenglei Ren <chenglei.ren@intel.com>

renchenglei commented:
Hi @tpalli & @YuanjunHuang, could you help take a review of this changes? The build issue may break cactus build, it is some tough for us.

tpalli commented:
Yes this makes sense to me, libmesa_anv_gen* depends on anv_extensions.h so it should be generated first. Note, there are some typos on the commit message, please fix those :)

It's a bit strange I never saw this though (and cannot reproduce) as android build system always does a paraller build.

renchenglei commented:
Thanks @tpalli. Updated the commit info.
For the build issue, it may be related to a more strict requirement on Android Q. And I could only reproduce this issue on some specific platform(APL Celadon).

strassek commented:
LGTM
I guess the plan for aosp is to have generated files prebuilt and commited:
https://android.googlesource.com/platform/external/mesa3d/+/c6638988f478690fac54a8a85d3c6367d190fb5d

renchenglei commented:
Thanks @strassek! 
I updated the commit title and added the "INTERNAL" flag. It seems this is one common issue for upstream mesa. Could we submit it to upstream?
@js0701 & @YuanjunHuang, could you help take a review of this and help to merge? We would better to get it landed on 1A, which could help unblock cactus team's engineering/daily/weekly/ build.

tpalli commented:
@renchenglei I've sent it upstream with you as author and a bit modified commit message here:
https://gitlab.freedesktop.org/mesa/mesa/merge_requests/940

js0701 commented:
LGTM. @renchenglei  Since Tapani has sent it to mailist, maybe we should just use https://gitlab.freedesktop.org/mesa/mesa/merge_requests/940
here.

renchenglei commented:
Thanks @tpalli & @js0701! I just updated the commit info align with https://gitlab.freedesktop.org/mesa/mesa/merge_requests/940
